crypto-ffi/src/wasm.rs

```diff
@@ -1122,40 +1113,22 @@ impl CoreCryptoWasmLogger {
             ctx,
         }
     }
-
-    fn clone_with_level(&self, level: &Level) -> Self {
-        Self {
-            logger: self.logger.clone(),
-            level: level.into(),
-            ctx: self.ctx.clone(),
-        }
-    }
 }
 
-impl std::io::Write for CoreCryptoWasmLogger {
-    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
-        let message = String::from_utf8_lossy(buf);
-        let _ = self
-            .logger
-            .call2(&self.ctx, &JsValue::from(self.level), &JsValue::from(message.as_ref()));
-        Ok(buf.len())
+impl log::Log for CoreCryptoWasmLogger {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        true
     }
 
-    fn flush(&mut self) -> std::io::Result<()> {
-        Ok(())
-    }
-}
-
-impl MakeWriter<'_> for CoreCryptoWasmLogger {
-    type Writer = Self;
-
-    fn make_writer(&self) -> Self::Writer {
-        self.clone()
+    fn log(&self, record: &Record) {
+        let message = format!("{}", record.args());
+        let level: CoreCryptoLogLevel = self.level.into();
+        self.logger
+            .call2(&self.ctx, &JsValue::from(level), &JsValue::from(message))
+            .unwrap();
```

Comment:
I can't return an error here so should I simply swallow the error?

Reply:
I'm not sure what kind of errors may pop up here, but we could perhaps consider using `web_sys::console` facilities to at least print out something in case this errors.

Reply:
Unsure about printing things to console I think it's best to swallow the error.

Reply:
Eh, the problem with swallowing errors is, if something unexpected happens, you're only going to see unexpected results on some higher layer, and the logs will be empty, leaving you wondering what really happened.

Admittedly, here it's a bit hard to come up with a scenario where this would fail, but OTOH, this does go through the wasm stack and if, for example, clients have a bug in their logging function, this is going to silently fail, and clients will wonder why they don't see anything in the logs. So it would be nice if the call fails here to dump the error to the JS console, perhaps it might even contain the traceback.

Reply:
But the error will be happening in the web app, they can catch and log the error themselves. Sure maybe they forget and it's will nice inform them about it.

I'll go ahead add the console log fallback.

Reply:
Ok, finally this this change and I added a test for it.